Sources/FoundationMacros/CMakeLists.txt

```diff
@@ -12,29 +12,47 @@
 ##
 ##===----------------------------------------------------------------------===##
 
+cmake_minimum_required(VERSION 3.22)
+
+if(POLICY CMP0156)
+    # Deduplicate linked libraries where appropriate
+    cmake_policy(SET CMP0156 NEW)
+endif()
+if(POLICY CMP0157)
+    # New Swift build model: improved incremental build performance and LSP support
+  cmake_policy(SET CMP0157 NEW)
+endif()
+
+project(FoundationMacros
+  LANGUAGES Swift)
+
+include(GNUInstallDirs)
```

Comment:
When testing this out for linux, I'm seeing:

```
CMake Warning (dev) at /Repos/swift-dev/cmake/Modules/GNUInstallDirs.cmake:243 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:29 (include)
```

from this include, are we missing something here?

Reply:
It seems that Swift does not count as a "language" for this module for some reason. That is a bug in CMake that we should address. For the time being, I think that this should be possible to workaround by explicitly passing in `-D CMAKE_INSTALL_LIBDIR=lib`. I think that we want to keep the include for the install paths.

Reply:
Yeah, we'll need to teach Swift to set up the `CMAKE_SYSTEM_NAME` and `CMAKE_SIZEOF_VOID_P`. In theory the frontend should know the size of a pointer (and endianness info), so I wonder if we couldn't stick that in the target info and pull it from there instead of trying to generate the tables in Swift? 🤔